base_layer/core/src/consensus/consensus_constants.rs

```diff
@@ -672,8 +672,33 @@ impl ConsensusConstants {
         &[OutputType::Coinbase, OutputType::Standard, OutputType::Burn]
     }
 
-    const fn current_permitted_range_proof_types() -> &'static [RangeProofType] {
-        &[RangeProofType::BulletProofPlus]
+    const fn current_permitted_range_proof_types() -> [(OutputType, &'static [RangeProofType]); 5] {
+        [
+            (OutputType::Standard, &[RangeProofType::BulletProofPlus]),
+            (OutputType::Coinbase, &[
+                RangeProofType::BulletProofPlus,
+                RangeProofType::RevealedValue,
+            ]),
+            (OutputType::Burn, &[
+                RangeProofType::BulletProofPlus,
+                RangeProofType::RevealedValue,
+            ]),
+            (OutputType::ValidatorNodeRegistration, &[
+                RangeProofType::BulletProofPlus,
+                RangeProofType::RevealedValue,
+            ]),
+            (OutputType::CodeTemplateRegistration, &[RangeProofType::BulletProofPlus]),
+        ]
+    }
+
```

Comment:
This must be reviewed and debated - this is an example. Easy to change the mapping that we want.

Reply:
General consensus: For now, we will only allow `OutputType::Coinbase` to have `RangeProofType::RevealedValue` as well; all other output types will only be `RangeProofType::BulletProofPlus`.